power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -583,6 +583,27 @@ template <symmetry_tag sym> struct NodeState {
     }
 };
 
+class RankIterator {
```

Comment:
`Iterator` is typically used as an object to iterate over. `RankIterator` may cause confusion. Maybe you can rename it to something like `RankedIteration`